File: python/1716.Calculate-Money-in-Leetcode-Bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        total = 0
        start = 1

        while n > 0:
            days = min(n, 7)

            for i in range(days):
                total += start + i
            
            n -= 7
            start += 1

        return total

File: javascript/1716.Calculate-Money-in-Leetcode-Bank.js
/**
 * @param {number} n
 * @return {number}
 */
var totalMoney = function (n) {
  let start = 1;
  let total = 0;

  while (n > 0) {
    let rem = n <= 7 ? n : 7;
    n = n - 7;
    let num = start;

    while (rem) {
      total += num;
      num += 1;
      rem -= 1;
    }

    start += 1;
  }

  return total;
};
